<launch>
    <param name="use_sim_time" value="true"/>

    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera_front_base" args="1.75, 0.0, 1.32 -0.0 0.21 -0.0 /base_link /camera_front_base 100" />
    <node pkg="tf" type="static_transform_publisher" name="camera_front_base_to_img" args="0.0, 0.0, 0.0 -1.5707963267949 0.0 -1.5707963267949 /camera_front_base /camera_front_img 100" />
    
    <node pkg="vision_lane_detection" type="ipm2" name="ipm2">
        <param name="image_transport" value="compressed"/>
		<param name="visualization_flag" value="true"/>
		<param name="odom_control" value="false"/>
		<param name="ipm_center_x" value="9.0"/>
		<param name="ipm_center_y" value="0.0"/>
		<param name="ipm_ROI_height" value="12.0"/>	
		<param name="ipm_ROI_near_width" value="4.0"/>
		<param name="ipm_ROI_far_width" value="12.0"/>
		<param name="publish_dis_thresh" value="0.2"/>
		<param name="scale" value="20.0"/>
		<param name="extract_training_image_ped_crossing" value="false"/>
		<param name="arrow_model_path" value="$(find vision_lane_detection)/data_and_model/arrow_20130308.model"/>
    	<param name="arrow_scale_path" value="$(find vision_lane_detection)/data_and_model/arrow_20130308.range"/>
		<param name="lane_model_path" value="$(find vision_lane_detection)/data_and_model/lane_20130305.model"/>
    	<param name="lane_scale_path" value="$(find vision_lane_detection)/data_and_model/lane_20130305.range"/>
		<param name="ped_crossing_model_path" value="$(find vision_lane_detection)/data_and_model/ped_crossing_20130305.model"/>
    	<param name="ped_crossing_scale_path" value="$(find vision_lane_detection)/data_and_model/ped_crossing_20130305.range"/>
    	<param name="road_roc_model_path" value="$(find vision_lane_detection)/data_and_model/roc_20130306.model"/>
    	<param name="road_roc_scale_path" value="$(find vision_lane_detection)/data_and_model/roc_20130306.range"/>
    	<param name="save_word_image" value="false"/>
    	<param name="lane_on" value="true"/>
    	<param name="arrow_on" value="true"/>
    	<param name="word_on" value="true"/>
    	<param name="crossing_on" value="true"/>
    	
    	<param name="visualize_arrow_info" value="true"/>
    	<param name="visualize_word_info" value="true"/>
	</node>

    <node pkg="image_view" type="image_view" name="image_view1">
        <remap from="image" to= "/camera_front/image_ipm"/>
        <param name="image_transport" value="compressed"/>
    </node>

    <node pkg="ocr_server" type="ocr_server" name="ocr_server">
    </node>

</launch>
